src/main/java/picard/analysis/TheoreticalSensitivity.java

```diff
@@ -204,4 +208,187 @@ public int draw() {
+     * @return Theoretical sensitivity for the given arguments at a constant depth.
+     */
+    public static double sensitivityAtConstantDepth(final int depth, final Histogram<Integer> qualityHistogram, final double logOddsThreshold, final int sampleSize, final double alleleFraction, final int randomSeed) {
+        final RouletteWheel qualityRW = new RouletteWheel(trimDistribution(normalizeHistogram(qualityHistogram)));
+        final Random randomNumberGenerator = new Random(randomSeed);
+        final BinomialDistribution bd = new BinomialDistribution(depth, alleleFraction);
+
+        // Calculate mean and deviation of quality score distribution to enable Gaussian sampling below
+        final double averageQuality = qualityHistogram.getMean();
+        final double standardDeviationQuality = qualityHistogram.getStandardDeviation();
+
+        int calledVariants = 0;
+        // Sample simulated variants, and count the number that would get called.  The ratio
+        // of the number called to the total sampleSize is the sensitivity.
+        for (int sample = 0; sample < sampleSize; sample++) {
+            final int altDepth = bd.inverseCumulativeProbability(randomNumberGenerator.nextDouble());
+
+            final int sumOfQualities;
+            if (altDepth < LARGE_NUMBER_OF_DRAWS) {
+                // If the number of alt reads is "small" we draw from the actual base quality distribution.
+                sumOfQualities = IntStream.range(0, altDepth).map(n -> qualityRW.draw()).sum();
+            } else {
+                // If the number of alt reads is "large" we draw from a Gaussian approximation of the base
+                // quality distribution to speed up the code.
+                sumOfQualities = drawSumOfQScores(altDepth, averageQuality, standardDeviationQuality, randomNumberGenerator.nextGaussian());
+            }
+
+            if (isCalled(depth, altDepth, (double) sumOfQualities, alleleFraction, logOddsThreshold)) {
+                calledVariants++;
+            }
+        }
+        return (double) calledVariants / sampleSize;
+    }
+
+    /**
+     * Simulates the sum of base qualities taken from reads that support the alternate allele by
+     * taking advantage of the fact that the sum of draws from a distribution tends towards a
+     * Gaussian per the Central Limit Theorem.
+     * @param altDepth Number of draws to take from base quality distribution
+     * @param averageQuality Average quality of alt bases
+     * @param standardDeviationQuality Sample standard deviation of base quality scores
+     * @param z number of standard deviation from the mean to take sum over
+     * @return Simulated sum of base qualities the support the alternate allele
+     */
+    static int drawSumOfQScores(final int altDepth, final double averageQuality, final double standardDeviationQuality, final double z) {
+        return (int) (altDepth * averageQuality + z * Math.sqrt(altDepth) * standardDeviationQuality);
```

Comment:
If and only if this is a speed bottleneck, you could easily make a static cache of integer square roots.

Reply:
In that case there's no need for a cache.